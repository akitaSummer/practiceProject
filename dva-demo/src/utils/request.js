import fetch from 'dva/fetch';
import qs from 'querystring'

function parseJSON(response) {
  return response.json();
}

function checkStatus(response) {
  if (response.status >= 200 && response.status < 300) {
    return response;
  }

  const error = new Error(response.statusText);
  error.response = response;
  throw error;
}

function handleHeader(options) {
  const headers = options.headers = options.headers ? options.headers : {}
  const defaultHeader = {
    'Content-type': 'application/x-www-form-urlencoded; charset=UTF-8'
  }

  options.headers = Object.assign({}, defaultHeader, headers)

  if (options.method === 'post') {
    let body = options.body ? options.body : {}
    body = qs.stringify(body)
    options.body = body
  }
}

/**
 * Requests a URL, returning a promise.
 *
 * @param  {string} url       The URL we want to request
 * @param  {object} [options] The options we want to pass to "fetch"
 * @return {object}           An object containing either "data" or "err"
 */
export default function request(url, options) {
  return fetch(url, options)
    .then(checkStatus)
    .then(parseJSON)
    .then(data => ({ data }))
    .catch(err => ({ err }));
}
