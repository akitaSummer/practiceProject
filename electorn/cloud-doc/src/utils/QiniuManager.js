const qiniu = require('qiniu')
const axios = require('axios')
const fs = require('fs')

class QiniuManager {
  constructor(accessKey, secretKey, bucket) {
    this.mac = new qiniu.auth.digest.Mac(accessKey, secretKey);
    this.bucket = bucket
    this.config = new qiniu.conf.Config();
    // 空间对应的机房
    this.config.zone = qiniu.zone.Zone_z0;
    this.bucketManager = new qiniu.rs.BucketManager(this.mac, this.config);
  }

  getBucketDomain() {
    const reqURL = `http://api.qiniu.com/v6/domain/list?tbl=${this.bucket}`
    const digest = qiniu.util.generateAccessToken(this.mac, reqURL)
    return new Promise((resolve, reject) => {
      qiniu.rpc.postWithoutForm(reqURL, digest, this._handleCallback(resolve, reject))
    })
  }

  downloadFile(key, downloadPath) {
    return this.generateDownloadLink(key).then(url => {
      const timestamp = new Date().getTime()
      url = `${url}?timestamp=${timestamp}`
      return axios({
        url,
        method: "GET",
        responseType: 'stream',
        header: { 'Cache-Control': 'no-cache' }
      })
    }).then(response => {
      const writer = fs.createWriteStream(downloadPath)
      response.data.pipe(writer)
      return new Promise((resolve, reject) => {
        writer.on('finish', resolve)
        writer.on('error', resolve)
      })
    }).catch(err => {
      return new Promise.reject({ err: err.response })
    })
  }

  generateDownloadLink(key) {
    this.publicBucketDomain = this.publicBucketDomain ? Promise.resolve([this.publicBucketDomain]) : this.getBucketDomain()
    return this.publicBucketDomain.then((arr) => {
      if (Array.isArray(arr) && arr.length > 0) {
        const reg = /^https?/
        const url = reg.test(arr[0]) ? arr[0] : `http://${arr[0]}`
        this.publicBucketDomain = url
        return this.bucketManager.publicDownloadUrl(this.publicBucketDomain, key)
      } else {
        throw Error('域名未找到，请查看储存空间是否过期')
      }
    })
  }

  _handleCallback(resolve, reject) {
    return (respErr, respBody, respInfo) => {
      if (respErr) {
        throw respErr;
      }
      if (respInfo.statusCode === 200) {
        resolve(respBody)
      } else {
        reject({
          statusCode: respInfo.statusCode,
          body: respBody
        })
      }
    }
  }

  uploadFile(key, localFilePath) {
    const options = {
      scope: this.bucket + ':' + key,
    };
    const putPolicy = new qiniu.rs.PutPolicy(options);
    const uploadToken=putPolicy.uploadToken(this.mac);
    const formUploader = new qiniu.form_up.FormUploader(this.config);
    const putExtra = new qiniu.form_up.PutExtra();
    // 文件上传
    return new Promise((resolve, reject) => {
      formUploader.putFile(uploadToken, key, localFilePath, putExtra, this._handleCallback(resolve, reject))
    })
  }

  deleteFile(key) {
    return new Promise((resolve, reject) => {
      this.bucketManager.delete(this.bucket, key, this._handleCallback(resolve, reject))
    })
  }

  getStat(key) {
    return new Promise((resolve, reject) => {
      this.bucketManager.stat(this.bucket, key, this._handleCallback(resolve, reject))
    })
  }
}

module.exports = QiniuManager
