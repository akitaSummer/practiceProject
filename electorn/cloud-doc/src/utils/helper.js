export const flattenArr = (arr) => {
  return arr.reduce((prev, item) => {
    prev[item.id] = item
    return prev
  }, {})
}

export const objToArr = (obj) => {
  return Object.keys(obj).map(key => obj[key])
}
