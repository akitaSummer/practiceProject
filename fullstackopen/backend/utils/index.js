const Blog = require('mongoose').model('Blog')

const dummy = (blogs) => {
  return 1
}

const totalLikes = async (blog) => {
  const findResult = await Blog.find({ name: blog.name })
  if (findResult.length === 0) {
    await new Blog(blog).save()
  }
  return (await Blog.find({})).length
}

module.exports = {
  dummy,
  totalLikes
}
