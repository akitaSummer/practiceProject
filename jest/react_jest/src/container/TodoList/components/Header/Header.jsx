import React, { useState } from 'react'

const Header = (props) => {

  const [value, setValue] = useState('')

  const handleInputKeyUp = (e) => {
    if (e.keyCode === 13 && value) {
      props.addUndoItem(value)
      setValue('')
    }
  }

  return (
      <div className='header'>
        <div className="header-content">
          Todo
          <input
            placeholder='Todo'
            className='header-input'
            type="text"
            data-test='input'
            value={value}
            onChange={(e) => setValue(e.target.value)}
            onKeyUp={handleInputKeyUp}
          />
        </div>
      </div>
  )
}

export default Header
