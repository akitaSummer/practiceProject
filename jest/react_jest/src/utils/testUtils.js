export const findTestWrapper = (wrapper, data) => {
  return wrapper.find(`[data-testid='${data}']`)
}
