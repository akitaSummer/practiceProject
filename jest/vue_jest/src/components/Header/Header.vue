<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
        class="header-input"
        type="text"
        data-testid="input"
        :value="inputValue"
        @input="e => changeValue(e.target.value)"
        @keyup.enter="addTodoItem"
        placeholder="TodoList"
      >
    </div>
  </div>
</template>
<script>
import { mapState, mapActions } from 'vuex'
export default {
  name: 'Header',
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.changeInputValue('')
      }
    },
    ...mapActions(['changeInputValue']),
    changeValue (value) {
      this.changeInputValue(value)
    }
  },
  computed: {
    ...mapState(['inputValue'])
  }
}
</script>

<style scoped lang="scss">
  .header {
    line-height: 60px;
    background: #333;
  }
  .header-content {
    width: 600px;
    margin: 0 auto;
    color: #fff;
    font-size: 24px;
  }
  .header-input {
    float: right;
    width: 360px;
    line-height: 24px;
    outline: none;
    margin-top: 16px;
    color: #333;
    text-indent: 10px;
  }
</style>
