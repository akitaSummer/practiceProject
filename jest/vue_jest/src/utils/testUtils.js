export const findTestWrapper = (wrapper, tag) => {
  return wrapper.findAll(`[data-testid='${tag}']`)
}
