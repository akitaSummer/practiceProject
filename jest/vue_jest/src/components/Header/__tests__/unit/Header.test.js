import { shallowMount } from '@vue/test-utils'
import { findTestWrapper } from '../../../../utils/testUtils'
import Header from '../../Header'

test('Header包含input框', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, 'input')
  expect(input.exists()).toBe(true)
})

test('input框发生变化，数据跟随改变', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('dell lee')
  const inputValue = wrapper.vm.inputValue
  expect(inputValue).toBe('dell lee')
})

test('input框无内容，输入回车无反应', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeFalsy()
})

test('input框有内容，输入回车触发事件, 同时清空inputValue', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('dell lee')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeTruthy()
  expect(wrapper.vm.inputValue).toBe('')
})

test('样式改变，做提示', () => {
  const wrapper = shallowMount(Header)
  expect(wrapper).toMatchSnapshot()
})
