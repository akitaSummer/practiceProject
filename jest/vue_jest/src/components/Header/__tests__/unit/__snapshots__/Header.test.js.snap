// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`样式改变，做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input type="text" data-testid="input" placeholder="TodoList" class="header-input"></div>
</div>
`;
